### DISCUSSION TITLE
How to test functions defined in *.sc file

### DISCUSSION THREAD

**Discussion by jiramares** [VirtusLab/scala-cli#2926] [Scripting]
Hi,
I am using scala-cli for scripting. I am creating runnable first.sc files and I would like to test function located in it. So I created in same directory first.test.scala.

first.sc:
```
#!/usr/bin/env -S shebang
println(sum(2, 3))
def sum(a: Int, b: Int) = a + b
```

first.test.scala:
```
//> using test.dep org.scalatest::scalatest::3.2.18

import org.scalatest.funsuite.AnyFunSuite

class FirstTest extends AnyFunSuite:
    test("Sum 2 + 4 is 6"):
        assert(first.sum(2, 4) == 6)
```

But the first (in first.sum(2, 4)) is not resolved

**Comment by Gedochao** [comment]
Hey, sorry for late answer.
I'm not sure what's the problem.

I have the 2 files you listed in a dir:
```bash
tree .
# .
# ├── first.sc
# └── first.test.scala
# 
# 1 directory, 2 files
```
The test is running as expected:
```bash
scala-cli test . 
# Compiling project (Scala 3.4.2, JVM (17))
# Compiled project (Scala 3.4.2, JVM (17))
# Compiling project (test, Scala 3.4.2, JVM (17))
# Compiled project (test, Scala 3.4.2, JVM (17))
# FirstTest:
# 5
# - Sum 2 + 4 is 6
# Run completed in 61 milliseconds.
# Total number of tests run: 1
# Suites: completed 1, aborted 0
# Tests: succeeded 1, failed 0, canceled 0, ignored 0, pending 0
# All tests passed.
```

@jiramares let me know how to reproduce the problem, in case I'm missing something.

**Comment by jiramares** [comment]
Hi, I am sorry it looks that I have tried to simplify example too much. 

First I don't know how to reference function in script if the script name contains '-' (minus). So, rename file `first.sc` to `fi-rst.sc` and 'first.test.scala` to 'fi-rst.test.scala`. How to ivoke function `sum` now? Neither `assert(fir-st.sum(2, 4) == 6` nor `assert(fir$minusst.sum(2, 4) == 6` work.

Second smaller problem is that the `*.sc` file is run during test (when tests are run the `first.sc` prints 5 to stdout.

jirka

**Comment by jiramares** [comment]
I am sorry I don't know why it closed so this comment is only to open.
